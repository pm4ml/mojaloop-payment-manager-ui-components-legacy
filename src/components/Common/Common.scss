@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.validation__icon {
  fill: transparent;
}

.validation__icon.validation__icon--invalid {
  height: auto;
  width: auto;
  fill: $color__main-danger;
}

.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 14px;
  padding-right: 18px;
  background: $color__gray-1;
  box-shadow: 0px 1px 2px $color__alpha-2;
  border-radius: 3px;
  min-width: 60px;
  display: block;
}

.validation__message {
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
  display: flex;
  align-items: center;
  font-size: 12px;
  color: $color__dark-gray-1;
  min-width: 60px;
}

.validation__message--inactive {
  text-decoration: line-through;
  stroke: none;
  color: $color__gray-5;
}

.validation__message__text {
  white-space: nowrap;
}

.validation__message__icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.validation__message__icon--active {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-danger;
  stroke: $color__main-danger;
}

.validation__message__icon--inactive {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-success;
}

.validation__message__icon--unset {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  height: 6px;
  width: 6px;
  margin: 3px;
  border-radius: 3px;
  display: flex;
  background: #ddd;
}

@keyframes fadeInValidationMessageIcon {
  0% {
    transform: scale(0.5, 0.5);
    opacity: 0.5;
  }

  100% {
    transform: scale(1, 1);
    opacity: 1;
  }
}

.named__icon {
  height: 20px;
  width: 20px;
  border-radius: 20px;
  font-size: 13px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #022843;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
