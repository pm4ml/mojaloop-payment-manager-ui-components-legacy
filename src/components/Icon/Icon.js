import './Icon.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Tooltip from '../Tooltip';

const iconSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
};

class Icon extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      name,
      fill,
      stroke,
      spin,
      tooltip,
      tooltipKind,
      tooltipDelay,
      tooltipPosition,
    } = this.props;

    const svgStyle = {
      height: `${size}px`,
      width: `${size}px`,
      fill,
      stroke,
      ...style,
    };
    const componentClassName = utils.composeClassNames([
      'el-icon',
      spin && 'el-icon--spin',
      className,
    ]);
    const svg = (
      <svg id={id} style={svgStyle} onClick={this.onClick} className={componentClassName}>
        <use xlinkHref={`#${name}`} />
      </svg>
    );
    if (tooltip) {
      return (
        <Tooltip label={tooltip} position={tooltipPosition} delay={tooltipDelay} kind={tooltipKind}>
          {svg}
        </Tooltip>
      );
    }
    return svg;
  }
}

Icon.propTypes = {
  id: PropTypes.string,
  style: PropTypes.shape(),
  className: PropTypes.string,
  size: PropTypes.number,
  name: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  spin: PropTypes.bool,
  onClick: PropTypes.func,
  tooltipKind: PropTypes.oneOf(['regular', 'error', 'info', 'warning', 'neutral']),
  tooltipDelay: PropTypes.number,
  tooltip: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
};
Icon.defaultProps = {
  id: undefined,
  style: undefined,
  className: undefined,
  size: 20,
  name: undefined,
  fill: undefined,
  stroke: undefined,
  spin: false,
  onClick: undefined,
  tooltipKind: undefined,
  tooltipDelay: undefined,
  tooltip: undefined,
  tooltipPosition: undefined,
};

export default Icon;
export { iconSizes };
