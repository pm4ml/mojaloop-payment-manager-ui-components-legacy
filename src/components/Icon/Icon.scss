@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
