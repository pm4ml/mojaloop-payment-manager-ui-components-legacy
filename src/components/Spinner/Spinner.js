import './Spinner.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

const polarToCartesian = (centerX, centerY, radius, angleInDegrees) => {
  const angleInRadians = (angleInDegrees - 90) * (Math.PI / 180.0);
  return {
    x: centerX + radius * Math.cos(angleInRadians),
    y: centerY + radius * Math.sin(angleInRadians),
  };
};

const describeArc = (x, y, radius, startAngle, endAngle) => {
  const start = polarToCartesian(x, y, radius, endAngle);
  const end = polarToCartesian(x, y, radius, startAngle);
  const largeArcFlag = endAngle - startAngle <= 180 ? '0' : '1';
  return ['M', start.x, start.y, 'A', radius, radius, 0, largeArcFlag, 0, end.x, end.y].join(' ');
};

class Spinner extends PureComponent {
  render() {
    const { size, center, color } = this.props;
    const realSize = typeof size === 'string' ? { s: 20, m: 40, l: 60 }[size] : size;
    const strokeWidth = realSize / 10;
    const width = `${realSize}px`;
    const height = `${realSize}px`;
    const position = realSize / 2;
    const radius = position - strokeWidth;
    const spinnerStyle = {
      width: center === false && width,
      height: center === false && height,
    };
    const pathStyle = { stroke: color };

    return (
      <div className={`el-spinner ${center ? 'center' : ''}`} style={spinnerStyle}>
        <svg
          className="el-spinner__component"
          width={width}
          height={height}
          viewBox={`0 0 ${realSize} ${realSize}`}
        >
          <path
            className="el-spinner__svg-path"
            strokeWidth={strokeWidth}
            d={describeArc(position, position, radius, 90, 200)}
            style={pathStyle}
          />
        </svg>
      </div>
    );
  }
}

Spinner.defaultProps = {
  size: 20,
  center: false,
  color: undefined,
};

Spinner.propTypes = {
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  center: PropTypes.bool,
  color: PropTypes.string,
};
export default Spinner;
