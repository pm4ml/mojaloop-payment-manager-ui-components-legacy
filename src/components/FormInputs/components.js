import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
