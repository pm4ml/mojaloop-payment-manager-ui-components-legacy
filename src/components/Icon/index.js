import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
