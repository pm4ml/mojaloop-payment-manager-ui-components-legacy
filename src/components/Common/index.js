import './Common.scss';
import '../../icons/mule/check-small.svg';
import '../../icons/mule/close-small.svg';

import PropTypes from 'prop-types';
import React from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Row from '../Row';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  let innerButtonSize;
  if (size === 'l') {
    innerButtonSize = 'm';
  } else if (size === 'm') {
    innerButtonSize = 's';
  } else {
    innerButtonSize = 'xs';
  }

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      size={innerButtonSize}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ActiveValidationIcon = () => (
  <Icon className="validation__message__icon--active" name="close-small" size={12} />
);
const InactiveValidationIcon = () => (
  <Icon className="validation__message__icon--inactive" name="check-small" size={14} />
);
const UnsetValidationIcon = () => <div className="validation__message__icon--unset" />;

const ValidationIcon = ({ active }) => {
  if (active === true) {
    return <ActiveValidationIcon />;
  } else if (active === false) {
    return <InactiveValidationIcon />;
  }
  return <UnsetValidationIcon />;
};

const ValidationMessage = ({ message, active }) => (
  <Row>
    <li
      className={`validation__message ${active === false ? 'validation__message--inactive' : ''}`}
    >
      <div className="validation__message__icon-container">
        <ValidationIcon active={active} />
      </div>
      <span className="validation__message__text">{message}</span>
    </li>
  </Row>
);

ValidationMessage.propTypes = {
  message: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  message: undefined,
  active: undefined,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map(({ message, active }, i) => (
      <ValidationMessage key={i.toString()} message={message} active={active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(ValidationMessage.propTypes)),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ size }) => (
  <Icon
    size={iconSizes[size]}
    name="warning-sign"
    className="mb-input__inner-icon validation__icon validation__icon--invalid "
    tooltip="Invalid"
    tooltipDelay={500}
    tooltipKind="error"
  />
);

const ValidationWrapper = ({ messages, active, children }) => (
  <Tooltip
    position="right"
    kind="neutral"
    custom
    delay={500}
    content={<ValidationMessages messages={messages} />}
    forceVisibility={active === true && messages && messages.length > 0}
    showOnHover={false}
  >
    {children}
  </Tooltip>
);

const NamedIcon = ({ namedIconTitle }) => <div className="named__icon">{namedIconTitle[0]}</div>;

NamedIcon.propTypes = {
  namedIconTitle: PropTypes.string,
};
NamedIcon.defaultProps = {
  namedIconTitle: undefined,
};

export { Loader, Placeholder, InnerButton, ValidationWrapper, InvalidIcon, NamedIcon };
