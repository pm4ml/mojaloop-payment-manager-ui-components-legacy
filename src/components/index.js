import '../assets/styles/index.scss';

import Button from './Button';
import Checkbox from './Checkbox';
import Column from './Column';
import ContentReader from './ContentReader';
import ControlIcon from './ControlIcon';
import DataList, { Link } from './DataList';
import DatePicker from './DatePicker';
import Dropdown from './Dropdown';
import FileUploader from './FileUploader';
import FormInput, { FormInputs, FormInputsGroup, Inline } from './FormInputs';
import Heading from './Heading';
import Icon from './Icon';
import Menu, { MenuItem, MenuSection } from './Menu';
import MessageBox from './MessageBox';
import Modal, { ModalTabsLayout } from './Modal';
import Navbar from './Navbar';
import Pill from './Pill';
import RadioGroup from './RadioGroup';
import Row from './Row';
import ScrollBox from './ScrollBox';
import Select from './Select';
import Spinner from './Spinner';
import { Tab, TabList, TabPanel, TabPanels, Tabs } from './Tabs';
import TextArea from './TextArea';
import TextField from './TextField';
import Toast from './Toast';
import Toggle from './Toggle';
import Tooltip from './Tooltip';

export {
  Button,
  Column,
  ContentReader,
  ControlIcon,
  Checkbox,
  DataList,
  DatePicker,
  Dropdown,
  FileUploader,
  FormInput,
  FormInputs,
  FormInputsGroup,
  Heading,
  Icon,
  Inline,
  Link,
  Menu,
  MenuSection,
  MenuItem,
  MessageBox,
  Modal,
  ModalTabsLayout,
  Navbar,
  Pill,
  RadioGroup,
  Row,
  ScrollBox,
  Select,
  Spinner,
  Tabs,
  Tab,
  TabList,
  TabPanels,
  TabPanel,
  TextArea,
  TextField,
  Toast,
  Toggle,
  Tooltip,
};
