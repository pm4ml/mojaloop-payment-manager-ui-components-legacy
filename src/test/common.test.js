import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { Loader, Placeholder } from '../components/Common';
import { iconSizes } from '../components/Icon';
import Spinner from '../components/Spinner';

describe('The Loader', () => {
  it('renders the spinner', () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper.find(Spinner).exists()).toBeTruthy();
  });

  it('renders the large, medium, small sizes', () => {
    const sizes = {
      large: 'l',
      medium: 'm',
      small: 's',
    };
    Object.entries(sizes).forEach(([, size]) => {
      const iconSize = iconSizes[size];
      const wrapper = shallow(<Loader size={size} />);
      const spinner = wrapper.find(Spinner);
      expect(spinner.prop('size')).toEqual(iconSize);
    });
  });

  it('renders the loader correctly when multiple props are set', () => {
    const wrapper = shallow(<Loader size="m" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('The Placeholder', () => {
  it('renders the placeholder', () => {
    const wrapper = shallow(<Placeholder />);
    expect(wrapper.is('label')).toEqual(true);
  });

  it('renders the label prop', () => {
    const wrapper = shallow(<Placeholder label="foo" />);
    const label = wrapper.find('label');
    expect(label.text()).toEqual('foo');
  });

  it('renders the large, medium, small sizes', () => {
    const sizes = {
      large: 'l',
      medium: 'm',
      small: 's',
    };
    Object.entries(sizes).forEach(([name, size]) => {
      const wrapper = shallow(<Placeholder size={size} />);
      const label = wrapper.find('label');
      const classname = `mb-input__placeholder--${name}`;
      expect(label.hasClass(classname)).toBe(true);
    });
  });

  it('renders the large size active sizes', () => {
    const sizes = {
      large: 'l',
      medium: 'm',
      small: 's',
    };
    Object.entries(sizes).forEach(([name, size]) => {
      const wrapper = shallow(<Placeholder active size={size} />);
      const label = wrapper.find('label');
      const classname = `mb-input__placeholder--active-${name}`;
      expect(label.hasClass(classname)).toBe(true);
    });
  });

  it('renders the default large size active status when size is not specified', () => {
    const wrapper = shallow(<Placeholder active />);
    const label = wrapper.find('label');
    expect(label.hasClass('mb-input__placeholder--active-large')).toEqual(true);
  });

  it('renders the placeholder correctly when multiple props are set', () => {
    const wrapper = shallow(<Placeholder label="foo" active size="m" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
