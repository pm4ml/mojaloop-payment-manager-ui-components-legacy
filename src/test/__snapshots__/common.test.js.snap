// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loader renders the loader correctly when multiple props are set 1`] = `
<div
  className="mb-input__inner-icon mb-loader"
>
  <Spinner
    center={false}
    size={14}
  />
</div>
`;

exports[`The Placeholder renders the placeholder correctly when multiple props are set 1`] = `
<label
  className="mb-input__placeholder mb-input__placeholder--medium mb-input__placeholder--active-medium"
>
  foo
</label>
`;
